clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -48,6 +57,19 @@ class ElasticsearchIndex[CI <: ClioIndex](
     s"$rootDir$dir/"
   }
 
+  /**
+    * The ID of a json record in elasticsearch. By default, each record's ID consists
+    * of the concatenated values of the Key fields.
+    * Additional fields can be added to the ID using the ElasticsearchIndex constructor.
+    */
+  def getId(json: Json): String = {
+    // Very hacky way of getting key fields without instantiating dummy keys
+    val keyFields = clioIndex.keyTag.runtimeClass.getConstructors.head.getParameters
```

Comment:
Would prefer to use the `FieldMapper` for the key type instead of the class tag. It's shapeless-fu instead of reflection.

Reply:
We should cache this calculation at the instance level too, instead of doing it on every `getId` call.